mkdocs/utils/__init__.py

```diff
@@ -53,7 +53,7 @@ def get_build_timestamp(*, pages: Collection[Page] | None = None) -> int:
     if pages:
         # Lexicographic comparison is OK for ISO date.
         date_string = max(p.update_date for p in pages)
-        dt = datetime.fromisoformat(date_string)
+        dt = datetime.fromisoformat(date_string).replace(tzinfo=timezone.utc)
```

Comment:
The issue is that we're using `utc` timezone datetimes everywhere, except here.
The robust fix is that we update this naive timezone into a `utc` based timezone.